File: javabackend/src/main/java/mft/dev/javabackend/dto/insert/UtenteInsertRequestDTO.java
package mft.dev.javabackend.dto.insert;

import com.fasterxml.jackson.annotation.JsonProperty;
import mft.dev.javabackend.model.Todo;
import mft.dev.javabackend.model.Utente;

import java.util.ArrayList;
import java.util.List;

public record UtenteInsertRequestDTO(
        @JsonProperty(value = "nome", required = true)
        String nome,
        @JsonProperty(value = "cognome", required = true)
        String cognome,
        @JsonProperty("email")
        String email,
        @JsonProperty("todos")
        List<TodoInsertRequestDTO> todosDto
) {
        public Utente toUtente(){
                Utente utente = new Utente();
                utente.setNome(this.nome());
                utente.setCognome(this.cognome());
                utente.setEmail(this.email());

                List<Todo> todos = todosDto() != null ? todosDto().stream().map(t -> t.toTodo(utente)).toList() : new ArrayList<>();
                utente.setTodoList(todos);

                return utente;
        }
}

File: kotlinbackend/src/main/kotlin/mft/dev/kotlinbackend/dto/insert/UtenteInsertRequestDTO.kt
package mft.dev.kotlinbackend.dto.insert

import com.fasterxml.jackson.annotation.JsonProperty
import mft.dev.kotlinbackend.model.Todo
import mft.dev.kotlinbackend.model.Utente

data class UtenteInsertRequestDTO(
    @JsonProperty("nome", required = true)
    val nome: String,
    @JsonProperty("cognome", required = true)
    val cognome: String,
    @JsonProperty("email")
    val email: String? = null,
    @JsonProperty("todos")
    val todosDto: MutableList<TodoInsertRequestDTO>?
) {
    companion object{

    }

    fun toUtente(): Utente{
        val utente = Utente(nome = nome, cognome = cognome, email = email)
        val todos: MutableList<Todo> = todosDto?.map{it.toTodo(utente)}?.toMutableList() ?: mutableListOf()
        utente.todoList = todos

        return utente
    }
}
